src/form.js

```diff
@@ -209,6 +217,7 @@ define([
         this.formName = 'New Form';
         this.mugMap = {};
         this.tree = new Tree('data', 'control');
+        xpath.addHashtag('#form', '/data');
```

Comment:
This is modifying global state. I understand that the global state it's modifying is an implementation detail of the xpath parser. But we should avoid building APIs that abuse that global state so it will be easy to port on the day that the xpath parser does not use global state (and then we can create an xpath parser instance with it's own local state for each form).

I suggest that we make a function like `form.addHashtag(hashtag, xpath)`, and that function will modify a `hashtagToXPath` object which is owned by the form. `hashtagToXPath` will be passed to `makeXPathModels()` and the result of that call is also owned by the form. That way all of the global state is left as only an implementation detail of the xpath parser library.